openwisp_users/api/views.py

```diff
@@ -209,28 +208,20 @@ def initial(self, *args, **kwargs):
         super().initial(*args, **kwargs)
         self.assert_parent_exists()
 
-    def assert_parent_exists(self):
-        try:
-            assert self.get_parent_queryset().exists()
-        except (AssertionError, ValidationError):
-            user_id = self.kwargs['pk']
-            raise NotFound(detail=_("User with ID '{}' not found.".format(user_id)))
-
     def get_parent_queryset(self):
-        user = self.request.user
-
-        if user.is_superuser:
-            return User.objects.filter(pk=self.kwargs['pk'])
-
-        org_users = OrganizationUser.objects.filter(user=user).select_related(
-            'organization'
-        )
-        qs_user = User.objects.none()
-        for org_user in org_users:
-            if org_user.is_admin:
-                qs_user = qs_user | org_user.organization.users.all().distinct()
-        qs_user = qs_user.filter(is_superuser=False)
-        return qs_user.filter(pk=self.kwargs['pk'])
+        qs = User.objects.filter(pk=self.kwargs['pk'])
+        if self.request.user.is_superuser:
+            return qs
```

Comment:
if the user performing the request is superuser, just return the parent without further checks (superusers can do anything).